eth/handler.go

```diff
@@ -698,8 +704,8 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		}
 
 	case msg.Code == TxMsg:
-		// Transactions arrived, make sure we have a valid chain to handle them
-		if atomic.LoadUint32(&pm.fastSync) == 1 {
+		// Transactions arrived, make sure we have a valid and fresh chain to handle them
+		if atomic.LoadUint32(&pm.fastSync) == 1 || atomic.LoadUint32(&pm.synced) == 0 {
```

Comment:
Do we still need the fast sync check? The fetcher doesn't run until fast sync is over, right?